src/test/rgw/test_d4n_filter.cc

```diff
@@ -2759,6 +2952,437 @@ TEST_F(D4NFilterFixture, CopyReplaceVersionedObjectWrite)
+    std::string oid = instance + "#0#" + std::to_string(ofs);
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + url_encode(bucketName, true) + "/" + TEST_OBJ + testName + "/" + oid, err), false);  
+
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + url_encode(bucketName, true) + "/" + TEST_OBJ + testName + "/" + version, err), false);  
+    oid = version + "#0#" + std::to_string(ofs);
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + url_encode(bucketName, true) + "/" + TEST_OBJ + testName + "/" + oid, err), false);  
+
+    conn->cancel();
+    testBucket->remove(env->dpp, true, optional_yield{yield});
+    DriverDestructor driver_destructor(static_cast<rgw::sal::D4NFilterDriver*>(driver));
+  }, rethrow);
+
+  io.reset();
+  io.run();
+}
+
+// Additional workflow tests
+TEST_F(D4NFilterFixture, SimpleDeleteBeforeCleaning)
+{
+  env->cct->_conf->d4n_writecache_enabled = true;
+  env->cct->_conf->rgw_d4n_cache_cleaning_interval = 0.5;
+  const std::string testName = "SimpleDeleteBeforeCleaning";
+  const std::string bucketName = "/tmp/d4n_filter_tests/dbstore-default_ns.1";
+  std::vector<std::string> instances;
+  std::string deleteMarker;
+ 
+  net::spawn(io, [this, &testName, &bucketName, &instances, &deleteMarker] (net::yield_context yield) {
+    init_driver(yield);
+    create_bucket(testName, yield);
+    testBucket->get_info().bucket.bucket_id = bucketName;
+    
+    for (int i = 0; i <= 1; ++i) {
+      std::string instance;
+      put_version_enabled_object(testName, instance, yield); // Upload two versions 
+      instances.push_back(instance);
+    }
+
+    std::unique_ptr<rgw::sal::Object::DeleteOp> del_op_enabled = objEnabled->get_delete_op();
+    objEnabled->set_instance(""); // Simple delete
+    EXPECT_EQ(del_op_enabled->delete_obj(env->dpp, optional_yield{yield}, rgw::sal::FLAG_LOG_OP), 0);
+
+    boost::system::error_code ec;
+    request req;
+    req.push("HGET", bucketName + "_" + TEST_OBJ + testName + "_0_0", "deleteMarker");
+
+    response< int > resp;
+
+    conn->async_exec(req, resp, yield[ec]);
+
+    ASSERT_EQ((bool)ec, false);
+    deleteMarker = std::get<0>(resp).value();
+
+    std::error_code err;
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + url_encode(bucketName, true) + "/" + TEST_OBJ + testName + "/" + deleteMarker, err), false);  
+
+    dynamic_cast<rgw::d4n::LFUDAPolicy*>(d4nFilter->get_policy_driver()->get_cache_policy())->save_y(null_yield);
+  }, rethrow);
+
+  io.run_for(std::chrono::seconds(2)); // Allow cleaning cycle to complete
+
+  net::spawn(io, [this, &testName, &bucketName, &deleteMarker] (net::yield_context yield) {
+    dynamic_cast<rgw::d4n::LFUDAPolicy*>(d4nFilter->get_policy_driver()->get_cache_policy())->save_y(optional_yield{yield});
+
+    std::error_code err;
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + url_encode(bucketName, true) + "/" + TEST_OBJ + testName + "/" + deleteMarker, err), false);  
```

Comment:
here you just need to do a simple get and see if the result returned is a delete marker, and check if the existing head block in the cache has been marked clean